packages/embeds/js/src/features/blocks/inputs/number/components/NumberInput.tsx

```diff
@@ -14,24 +16,22 @@ type NumberInputProps = {
 };
 
 export const NumberInput = (props: NumberInputProps) => {
-  const [inputValue, setInputValue] = createSignal<string | number>(
-    props.defaultValue ?? "",
-  );
-  const [staticValue, bindValue, targetValue] = numberInputHelper(() =>
-    inputValue(),
-  );
+  const numberInput = useNumberInput({
+    formatOptions: {
+      style: props.block.options?.style,
+      currency: props.block.options?.currency,
+      unit: props.block.options?.unit,
+    },
+  });
   let inputRef: HTMLInputElement | undefined;
 
-  const checkIfInputIsValid = () =>
-    inputRef?.value !== "" && inputRef?.reportValidity();
-
   const submit = () => {
-    if (checkIfInputIsValid())
+    if (!numberInput().invalid) {
       props.onSubmit({
         type: "text",
-        value: inputRef?.value ?? inputValue().toString(),
+        value: numberInput().valueAsNumber.toFixed(DEFAULT_PRECISION),
```

Comment:
Why not:

```suggestion
        value: numberInput().valueAsNumber.toString(),
```

So that we don't have to provide a precision?